packages/contracts-bedrock/test/setup/FFIInterface.sol

```diff
@@ -245,6 +245,56 @@ contract FFIInterface {
         return (memRoot, proof);
     }
 
+    function getCannonMemoryProof(
+        uint32 pc,
+        uint32 insn,
+        uint32 memAddr,
+        uint32 memVal,
+        uint32 memAddr2,
+        uint32 memVal2
+    )
+        external
+        returns (bytes32, bytes memory)
+    {
+        string[] memory cmds = new string[](9);
+        cmds[0] = "scripts/go-ffi/go-ffi";
+        cmds[1] = "diff";
+        cmds[2] = "cannonMemoryProof";
+        cmds[3] = vm.toString(pc);
+        cmds[4] = vm.toString(insn);
+        cmds[5] = vm.toString(memAddr);
+        cmds[6] = vm.toString(memVal);
+        cmds[7] = vm.toString(memAddr2);
+        cmds[8] = vm.toString(memVal2);
+        bytes memory result = Process.run(cmds);
+        (bytes32 memRoot, bytes memory proof) = abi.decode(result, (bytes32, bytes));
+        return (memRoot, proof);
+    }
+
+    function getCannonMemoryProof2(
```

Comment:
The distinction between these 2 methods is a little opaque - maybe worth some comments here. 

I also wonder if we even need this second style ("cannonMemoryProof2").  Seems like we could instead have the original method return the full proof (insn + mem1 + mem2 proofs), and then add another method that breaks up that full proof into those 3 constituent proofs?

Reply:
I originally went with a single function to do this. But the api is awkward as it needs to support building a tree containing either two or three non-zero nodes, while generating a proof for another, possible disjoint zero node.
With the existing `cannonMemoryProof`, you'd have to support these two use cases:
```
addressA valueA addressB valueB proveAddressC
```
and
```
addressA valueA proveAddressB
```
It's not straight forward to disambiguat the two without introducing a flag, which is one option.

I figured it'll be clearer to have a separate method for this usecsse.